types/sharedb/index.d.ts

```diff
@@ -346,3 +350,8 @@ interface PresenceMessage {
 }
 
 type BasicCallback = (err?: Error) => void;
+
+type ErrorHandler = (error: Error, context: ErrorHandlerContext) => void;
+interface ErrorHandlerContext {
+    agent?: Agent;
```

Comment:
🤔  Now that I see the typedef written out, I'm starting to think it'd be useful to add a "source" or "type" field to the context.

Consumers could look at it to determine where the error came from, and then TS's discriminated union analysis eliminates the need to separately check the shape of the context field.

Adding a field to the context would be backwards-compatible, so going with this for now is fine by me.

Reply:
Isn't that what the stack trace is for? (We should probably move this conversation into `sharedb`)